subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/VerifyArtifactsStep.kt

```diff
@@ -51,13 +51,12 @@ open class VerifyArtifactsStep(
             entries.set(project.files(artifactsConfig.outputs.values.map { it.path }))
 
             project.gradle.buildFinished { buildResult ->
-                val isCopyScheduled = buildResult.gradle?.isTaskScheduled(this) ?: false
-                // TODO: consider skipped state (--dry-run)
-                val copyFinished = this.didWork
-                if (isCopyScheduled && !copyFinished) {
-                    project.ciLogger.info("Artifacts copying was scheduled but didn't run. Start it now")
-                    if (!this.didWork) {
-                        this.doAction()
+                if (buildResult.failure != null && buildResult.action == "Build") {
```

Comment:
It's good to hide this low-level knowledge about "action == "Build".

Reply:
Hide where?

Reply:
For example, in an extension utility method.

We missed this one or two times already. `buildFinished` callback can be called multiple times. To distinguish a "real" build we've found only this workaround.

For me, the problem is similar to [a main process](https://github.com/int02h/primaree/blob/master/library/src/main/java/com/dpforge/primaree/ApplicationExtensions.kt) - workaround for lifecycle.

Reply:
Done here - 3da553495ac0194d7b1ae822e0d163d6d4af731a